src/index/LocatedTriples.h

```diff
@@ -130,32 +130,36 @@ class LocatedTriplesPerBlock {
   // 1. The `locatedTriples` must not already exist in
   // `LocatedTriplesPerBlock`.
   std::vector<LocatedTriples::iterator> add(
-      const std::vector<LocatedTriple>& locatedTriples);
+      std::span<const LocatedTriple> locatedTriples,
+      std::vector<CompressedBlockMetadata> originalMetadata);
 
-  void erase(size_t blockIndex, LocatedTriples::iterator iter);
+  void erase(size_t blockIndex, LocatedTriples::iterator iter,
+             std::vector<CompressedBlockMetadata> originalMetadata);
 
   // Get the total number of `LocatedTriple`s (for all blocks).
   size_t numTriples() const { return numTriples_; }
 
   // Get the number of blocks with a non-empty set of located triples.
   size_t numBlocks() const { return map_.size(); }
 
-  // Update the augmented block metadata. Must be called after adding/removing
-  // triples.
-  void updateAugmentedMetadata(
-      const std::vector<CompressedBlockMetadata>& metadata);
+  // Must be called initially before using the `LocatedTriplesPerBlock` to
+  // initialize the block metadata that is augmented for updated triples. This
+  // is currently done in `Permutation::loadFromDisk`.
+  void updateAugmentedMetadata(std::vector<CompressedBlockMetadata> metadata);
```

Comment:
Is that comment in sync? 
I still see a lot of `originalMetadata` as arguments.
What about storing two metadata: The original one, and the current one.
Then you hgave to initialize them, but you don't need to pass them as arguments to `add`. This seems to be the cleaner design for me.

Reply:
It was in sync, yes. I applied your suggestion.